app/controllers/coronavirus_form/expert_advice_type_controller.rb

```diff
@@ -4,30 +4,30 @@ class CoronavirusForm::ExpertAdviceTypeController < ApplicationController
   TEXT_FIELDS = %w(expert_advice_type_other).freeze
 
   def submit
-    expert_advice_type = Array(params[:expert_advice_type]).map { |item| strip_tags(item).presence }.compact
-    expert_advice_type_other = strip_tags(params[:expert_advice_type_other]).presence
-    session[:expert_advice_type] = expert_advice_type
-    session[:expert_advice_type_other] = if selected_other?(expert_advice_type)
-                                           expert_advice_type_other
-                                         else
-                                           ""
-                                         end
+    @form_responses = {
+      expert_advice_type: Array(params[:expert_advice_type]).map { |item| strip_tags(item).presence }.compact,
+    }
+    @form_responses[:expert_advice_type_other] = strip_tags(params[:expert_advice_type_other]).presence if selected_other?(@form_responses[:expert_advice_type])
```

Comment:
Now that an instance variable is being used, does `@form_responses[:expert_advice_type]` still need to be passed to `selected_other?`

Reply:
It doesn't.  Have removed.